code_issues.qmd

```diff
@@ -301,6 +301,23 @@ Packages can write per default to `tempdir()`, or any other temporary/cache dire
 
 <!-- tbd: add section about R_user_dir() --> 
 
+---------------------------------------
+
+# Leaving files in the temporary directory
+
+## Problem
+
+Some of your code that runs during ` R CMD check ` creates files in the temporary directory specified by the environment variable `TMPDIR` and doesn't remove them afterwards. As a result, ` R CMD check ` gives your package a NOTE complaining about the "detritus in the temp directory".
+
+## Solution
+
+If you use `tempfile()`/`tempdir()` as the default destination for functions called in your examples or tests, `unlink()` the resulting files as part of the test. In the `\examples{}` section of your help files, clean-up actions can be wrapped in `\dontshow{}` for aesthetic reasons. If you launch a browser, Calibre, or Python code, and it creates temporary files, find out what they are and remove them once the launched code is done with them.
```

Comment:
Perhaps where instead of what:

```suggestion
If you use `tempfile()`/`tempdir()` as the default destination for functions called in your examples or tests, `unlink()` the resulting files as part of the test. In the `\examples{}` section of your help files, clean-up actions can be wrapped in `\dontshow{}` for aesthetic reasons. If you launch a browser, Calibre, Python code or other software, and it creates temporary files, find out where they are and remove them once the code is done with them.
```